DIG-898: Show username of the authenticated user in data portal

OrdiNeu opened the pull request:
## Ticket(s)

- [DIG-898](Show username of the authenticated user in data portal)

## Description

- This adjusts the profile button at the top right to show the user key of the currently-signed-in user (usually their email address). **This requires the corresponding branch from Query [here](https://github.com/CanDIG/candigv2-query/pull/42)**

## Screenshots (if appropriate)

### Before PR

### After PR
![image](https://github.com/CanDIG/candig-data-portal/assets/4656440/7db7ecba-54a8-4550-8cf8-c73899d1d285)

## To do/Tickets to be made before merging branch

- Requires the Query PR [here](https://github.com/CanDIG/candigv2-query/pull/42)

## Types of Change(s)

-   [x] 🪲 Bug fix (non-breaking change that fixes an issue)
-   [x] ✨ New feature (non-breaking change that adds functionality)
-   [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Has it been tested for:

-   [ ] My change requires a change to the documentation
-   [ ] I have updated the documentation accordingly
-   [ ] Prettier linter doesn't return errors
-   [ ] Production branch PR browser testing: Chrome, Firefox, Edge, etc.
-   [ ] Locally tested
-   [ ] Dev server tested
-   [ ] Production tested when merging into stable/production branch
-   [ ] [Runbook tasks](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) pass locally/on UHN-Dev
-   [ ] If visuals have changed, [Runbook](https://candig.atlassian.net/wiki/spaces/CA/pages/822018050/Frontend+Testing+Runbook) has been updated with new screenshots


[DIG-898]: https://candig.atlassian.net/browse/DIG-898?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

OrdiNeu committed:
Add the ability to check your own username from the frontend

OrdiNeu committed:
Adjust styling of the profile button

OrdiNeu committed:
Remove unused import

CourtneyGosselin reviewed (changes requested):
The pop-ups appear in the wrong location, likely depending on where you click the settings toggle. Additionally, the email/username (site_admin@test.ca) is not displayed.
![image](https://github.com/CanDIG/candig-data-portal/assets/37649170/0d33ccd1-bda6-4fcd-a051-7e39bc6dc043) Furthermore, when logging out, I encountered this error message:
![image](https://github.com/CanDIG/candig-data-portal/assets/37649170/523d2d28-5ab6-4ff1-9bea-3d03cbe602b5)
Attempting to log out again resulted in the following:
![image](https://github.com/CanDIG/candig-data-portal/assets/37649170/9b0ffe3e-922c-4cce-a725-5f4a055170b1)

I am unable to log out.

Additionally, it may be worth addressing the overflow issue in the settings module:
![image](https://github.com/CanDIG/candig-data-portal/assets/37649170/a718b8ff-6026-43f1-b21a-33d3f0a9cf6c)

OrdiNeu committed:
Remove overflow scrollbar

OrdiNeu committed:
Fixup profile popper anchor element behaviour

OrdiNeu commented:
> The pop-ups appear in the wrong location, likely depending on where you click the settings toggle.

Fixed in [28f9926](https://github.com/CanDIG/candig-data-portal/pull/153/commits/28f9926d7d65094b586a7254f3d81cff5e436fe3)

> Additionally, the email/username ([site_admin@test.ca](mailto:site_admin@test.ca)) is not displayed. ![image](https://private-user-images.githubusercontent.com/37649170/338336570-0d33ccd1-bda6-4fcd-a051-7e39bc6dc043.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NDY3NzMsIm5iZiI6MTcxODY0NjQ3MywicGF0aCI6Ii8zNzY0OTE3MC8zMzgzMzY1NzAtMGQzM2NjZDEtYmRhNi00ZmNkLWEwNTEtN2UzOWJjNmRjMDQzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE3VDE3NDc1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlNjVjNTFjY2EyYTdlNWQ1MzczN2Q5YTdlOWNiMDE2ZWJmOTgxODhkZmNjMTFkNTA2NjllNGJiMGE4MWFlY2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NCvSmsFK2niXa8A_K7G2yTWFkaDUI5IebQKB9cjK3sM) 

This can only happen if you have the wrong version of Query checked out -- it should be `feature/whoami`. Otherwise you'll see a 404 in your Network tab.

> Furthermore, when logging out, I encountered this error message: ![image](https://private-user-images.githubusercontent.com/37649170/338336761-523d2d28-5ab6-4ff1-9bea-3d03cbe602b5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NDY3NzMsIm5iZiI6MTcxODY0NjQ3MywicGF0aCI6Ii8zNzY0OTE3MC8zMzgzMzY3NjEtNTIzZDJkMjgtNWFiNi00ZmYxLTliZWEtM2QwM2NiZTYwMmI1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE3VDE3NDc1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA1OTRiZTc3NmZkMmNlNjY5NWRjMWU1NTlhZWViZmVmY2RkODVlZWI5ZTA3MzY5NjViMjRmOTRlOGY4YzNmYmQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.V04bLuvOA1_pJHWC3yr2txtNHe4K-2rOLie1j_-pIbU) Attempting to log out again resulted in the following: ![image](https://private-user-images.githubusercontent.com/37649170/338337147-9b0ffe3e-922c-4cce-a725-5f4a055170b1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NDY3NzMsIm5iZiI6MTcxODY0NjQ3MywicGF0aCI6Ii8zNzY0OTE3MC8zMzgzMzcxNDctOWIwZmZlM2UtOTIyYy00Y2NlLWE3MjUtNWY0YTA1NTE3MGIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE3VDE3NDc1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTcyYTJhYmVkMmEyMmZlZDdiMzg4MTQ5ZjMwNmZjZjViOTcwMWJkNmY4YjhiN2FmZmQwM2Q0OTI2YjQ2NzVkYWEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.vO1XrCxxy6vtkvrk0qDTpDTx6fprTXHvlZhy2vWsOzk)
> 
> I am unable to log out. 

This is not fixable here -- it's related to [these changes in Tyk](https://github.com/CanDIG/CanDIGv2/blame/develop/lib/tyk/configuration_templates/virtualLogout.js#L32), which were introduced but not caught [here](https://github.com/CanDIG/CanDIGv2/pull/513)

> Additionally, it may be worth addressing the overflow issue in the settings module: ![image](https://private-user-images.githubusercontent.com/37649170/338337809-a718b8ff-6026-43f1-b21a-33d3f0a9cf6c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2NDY3NzMsIm5iZiI6MTcxODY0NjQ3MywicGF0aCI6Ii8zNzY0OTE3MC8zMzgzMzc4MDktYTcxOGI4ZmYtNjAyNi00M2YxLWIyMWEtMzNkM2YwYTljZjZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE3VDE3NDc1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU3ZDA0ZTUwYWIyYTg5Y2JkZGIxNjNlODliYWVhNjA1NWNkNDI2MjM5ODMyNDc2MWE1MmE1Y2IxODYxZDQzZTUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.413PeS9lR3aAIQq9cZb-8292MVxhEkhPctnksrBvuOg)

Fixed in [f042108](https://github.com/CanDIG/candig-data-portal/pull/153/commits/f042108e43cc94c3962a6ebff0397114fa141fc1)

CourtneyGosselin reviewed (approved):
LGTM works as expected
